File: android/app/src/main/java/com/fuggel/Uway/model/Warning.kt
package com.fuggel.Uway.model

import com.fuggel.Uway.constants.Constants

data class Warning(
    val type: String?,
    val state: String?,
    val text: String?,
    val tts: String?,
    val eventWarningType: String?,
)

object WarningLogic {
    fun isValid(warning: Warning): Boolean {
        return !warning.type.isNullOrBlank() &&
                !warning.state.isNullOrBlank() &&
                !warning.text.isNullOrBlank() &&
                !warning.tts.isNullOrBlank() &&
                Constants.VALID_WARNING_TYPES.contains(warning.type) &&
                Constants.VALID_WARNING_STATES.contains(warning.state)
    }
}

File: ios/Models/Warning.swift
//
//  Warning.swift
//  Uway
//
//  Created by Furkan Ceylan on 19.06.25.
//

import Foundation

struct Warning {
    let type: String?
    let state: String?
    let text: String?
    let tts: String?
}

struct WarningLogic {
    static let validTypes = ["incident", "speed-camera"]
    static let validStates = ["early", "late"]

    static func isValid(_ warning: Warning) -> Bool {
        guard let type = warning.type,
              let state = warning.state,
              let text = warning.text,
              let tts = warning.tts else { return false }

        return validTypes.contains(type) && validStates.contains(state)
    }
}
